ath10k-ct: patches cleanup and firmwares update

Noltari opened the pull request:
More info: http://lists.infradead.org/pipermail/openwrt-devel/2020-April/023055.html

Noltari committed:
ath10k-ct: remove old patches

Only keep 5.4 patches.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

hnyman commented:
@Noltari 
Would it be possible to create package versioning for the ath10k-ct firmware?

Right now it is pretty confusing, as there is no change in the package version although the -ct firmware changes. The ath10k-firmware has its versioning based on the mainline ath10k version, and ath10k-ct firmware has no impact in the version. Despite your PR today, the ath10k-ct-firmware package would still be "ath10k-firmware-qca9984-ct_2019-10-03-d622d160-1"

My proposal would be to either:
* separate the ath10k-ct-firmware to its own package, so that it could be named reasonably (2020-04-24-v18 or something similar)
* create separate versioning inside the ath10k-firmware for the -ct firmware variants.

Noltari committed:
ath10k-firmware: move CT firmwares to new package

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari committed:
ath10k-ct-firmware: update firmware images

No release notes this time.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari committed:
ath10k-ct-firmware: rename ct-htt packages

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari committed:
ath10k-ct-firmware: add htt-mgt variants

For wave-2, there is now a new variant: htt-mgt-community (vs the old
full-htt-mgt-community).

The non-full one (hence forth 'diet') compiles out a lot of firmware features
that ath10k does not use. This saves a lot of resources and lets one
configure more stations/vdevs/etc using fwcfg.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari commented:
> @Noltari
> Would it be possible to create package versioning for the ath10k-ct firmware?
> 
> Right now it is pretty confusing, as there is no change in the package version although the -ct firmware changes. The ath10k-firmware has its versioning based on the mainline ath10k version, and ath10k-ct firmware has no impact in the version. Despite the actual version bump by your PR today, the ath10k-ct-firmware package would still be called "ath10k-firmware-qca9984-ct_2019-10-03-d622d160-1"
> 
> My proposal would be to either:
> 
> * separate the ath10k-ct-firmware to its own package, so that it could be named reasonably (2020-04-24-v18 or something similar)
> * create separate versioning inside the ath10k-firmware for the -ct firmware variants.

Done :)
